module_2/gift_certificates_system/service/src/main/java/com/epam/esm/service/TagService.java

```diff
@@ -0,0 +1,23 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dao.impl.TagDaoImpl;
+import com.epam.esm.entity.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ *  Tag SERVICE layer.
+ */
+@Component
```

Comment:
Why did you annotate abstarct class as component? 
Plus, please use specific annotations for services and repositories